apps/yggdrasil/priv/repo/migrations/20160229163000_create_role_resource.exs

```diff
@@ -0,0 +1,17 @@
+defmodule Yggdrasil.Repo.Migrations.CreateRoleResource do
+  use Ecto.Migration
+
+  def change do
+    create table(:roles_resources) do
+      add :role_id, references(:roles, on_delete: :nothing)
+      add :resource_id, references(:resources, on_delete: :nothing)
+      add :permission_id, references(:permissions, on_delete: :nothing)
+
+      timestamps
+    end
+    create index(:roles_resources, [:role_id])
+    create index(:roles_resources, [:resource_id])
+    create index(:roles_resources, [:permission_id])
```

Comment:
Yeah, that's what I was thinking, was copying what the generators do by default, but it doesn't make sense to have them and found it odd for the generators to do that, unless I just did it by accident on the character table, but I'm pretty sure that was generated.

And yes it does have the ID column as it's what ecto does by default and how the has_many and other relationships work by default. I was thinking about dropping it because role_id would suffice, as these really should not exist without a role.

I'm not sure why a composite key on all three though, I could see role and resource, but probably never look up a single record. Unless it's purely for the ordering. Could expand on that bit a little?